### DISCUSSION TITLE
2k  devtest  not  save file and get it

### DISCUSSION THREAD

**Discussion by alanmoleapfive** [filecoin-project/lotus#8260] [Storage Provider] [answered]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [X] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [X] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [X] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
Daemon:  1.14.4+2k+git.5b33555fd+api1.5.0
Local: lotus version 1.14.4+2k+git.5b33555fd

Daemon:  1.14.4+2k+git.5b33555fd+api1.4.0
Local: lotus-miner version 1.14.4+2k+git.5b33555fd
```

### Describe the Bug

wait  

2022-03-08T06:04:13.988Z        INFO    main    lotus-miner/init.go:748 Pushed CreateMiner message: bafy2bzacecybks5nb65oprkfnketzc4a774r3ppvya4c2yuw74t7kibx2akpe
2022-03-08T06:04:13.993Z        INFO    main    lotus-miner/init.go:749 Waiting for confirmation

lotus client list-deals
DealCid      DealId  Provider  State                     On Chain?  Slashed?  PieceCID     Size    Price            Duration  Verified  
...3q3uopfu  0       t01000    StorageDealFundsReserved  N          N         ...5qtonkdq  1016 B  0.000260639 FIL  521278    false     
root@0d3872e3afdd:/home/lotus# lotus-miner storage-deals list -v
Creation  Verified  ProposalCid  DealId  State  Client  Size  Price  Duration  TransferChannelID  Message
root@0d3872e3afdd:/home/lotus# lotus client list-deals
DealCid      DealId  Provider  State                     On Chain?  Slashed?  PieceCID     Size    Price            Duration  Verified  
...3q3uopfu  0       t01000    StorageDealFundsReserved  N          N         ...5qtonkdq  1016 B  0.000260639 FIL  521278    false     
root@0d3872e3afdd:/home/lotus# lotus state search-msg bafy2bzacealsmph4t3ta54zww6rwcyrdbdu276jjrx5wncka4xaiqawi7c6qa
Executed in tipset: [bafy2bzaceducqnrrl42varmavjvqewqwo44qqcxlatyvt7vhjchho6lxrlcws]
Exit Code: 0
Gas Used: 5014527
Return: 
root@0d3872e3afdd:/home/lotus# lotus client find bafy2bzacealsmph4t3ta54zww6rwcyrdbdu276jjrx5wncka4xaiqawi7c6qa
ERROR: no import found for root bafy2bzacealsmph4t3ta54zww6rwcyrdbdu276jjrx5wncka4xaiqawi7c6qa: no client blockstore for root bafy2bzacealsmph4t3ta54zww6rwcyrdbdu276jjrx5wncka4xaiqawi7c6qa

root@0d3872e3afdd:/home/lotus# lotus client retrieve bafy2bzacealsmph4t3ta54zww6rwcyrdbdu276jjrx5wncka4xaiqawi7c6qa hello2.txt
Failed to find file
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x17d5102]

goroutine 1 [running]:
github.com/filecoin-project/lotus/cli.glob..func33(0xc00083b440)
        /home/alan/lotus/cli/client_retr.go:321 +0x502
github.com/urfave/cli/v2.(*Command).Run(0x568cc60, 0xc000701e40)
        /home/alan/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/command.go:164 +0x64a
github.com/urfave/cli/v2.(*App).RunAsSubcommand(0xc000292300, 0xc000701ac0)
        /home/alan/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:427 +0x9a6
github.com/urfave/cli/v2.(*Command).startApp(0x568d0e0, 0xc000701ac0)
        /home/alan/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/command.go:279 +0x71b
github.com/urfave/cli/v2.(*Command).Run(0xc0006e5680, 0x7ffe4445d856)
        /home/alan/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/command.go:94 +0x7d9
github.com/urfave/cli/v2.(*App).RunContext(0xc000292000, {0x3eed3b0, 0xc000054050}, {0xc00004e0a0, 0x5, 0x5})
        /home/alan/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:306 +0x926
github.com/urfave/cli/v2.(*App).Run(...)
        /home/alan/go/pkg/mod/github.com/urfave/cli/v2@v2.2.0/app.go:215
github.com/filecoin-project/lotus/cli.RunApp(0x30e51c0)
        /home/alan/lotus/cli/helper.go:35 +0x4e
main.main()
        /home/alan/lotus/cmd/lotus/main.go:109 +0x8ec

### Logging Information

```text
gs-unverifiedbs
ffi-wrapper
peerstore
lock
repo
beacon
bitswap_network
dt-chanmon
providerstates
panic-reporter
ffiwrapper
builder
types
sub
vm
dht
```

### Repo Steps

1. Run '...'
2. Do '...'
3. See error '...'
...

**Comment by Reiers** [comment]
Hi @alanmoleapfive 

Moving to Discussions for help and troubleshooting!

Thanks for taking the time to create this ticket, but your issue lies elsewhere.

**Comment by LexLuthr** [comment, marked as answer]
@alanmoleapfive Hey, I think an important step was missed during the deal making process. The output CID you get from the `lotus client deal` command is a deal-proposal CID. This deal has to be accepted by the miner. I think this step was not executed.

`lotus-miner storage-deals pending-publish`
`lotus-miner storage-deals pending-publish --publish-now`

Once this is done, the state should change from "StorageDealFundsReserved" to accepted and then data-transfer and so on till sealing is complete. 

After that you can retrieve the file but again here you are using the CID of the deal proposal and thus getting an error. You will need to provide the CID of the file you want to retrieve. This is the same CID you provided as input in `lotus client deal` command.

Please execute all these steps and let us know if you still have a problem.

**Reply by alanmoleapfive** [reply]
Hi,LexLuthr
try do it
1、import data
# create file
head -c 450 /dev/urandom > 450_random_data.txt 
#import
 lotus client import ./450_random_data.txt
Import 1646710431103465273, Root bafk2bzaced6dqiehjujukwcdr56o7z5oi6d5iuj6j54uueslcdut25ixf43jo

# show <Data CID>
 lotus client local
1646710431103465272: bafk2bzaceaf3pam3lgllm5oddgwuzgu6auiqm2xzsirjbx6vwqfuq5mptp3jg @/home/lotus/512_random_data.txt (import)
1646710431103465273: bafk2bzaced6dqiehjujukwcdr56o7z5oi6d5iuj6j54uueslcdut25ixf43jo @/home/lotus/450_random_data.txt (import)

2、query-ask
lotus client query-ask t01000
Ask: t01000
Price per GiB: 0.0000000005 FIL
Verified Price per GiB: 0.00000000005 FIL
Max Piece size: 2 KiB
Min Piece size: 256 B

3、send deal
lotus client deal <Data CID> <miner> <price> <duration>
lotus client deal bafk2bzaced6dqiehjujukwcdr56o7z5oi6d5iuj6j54uueslcdut25ixf43jo t01000 0.0000000005  518400
bafyreigovkhfbhy6yl3xeshm5f7d67stigyfwn6u67ybiddhtdc2m26fvy

4、check
lotus client list-deals
DealCid      DealId  Provider  State                     On Chain?  Slashed?  PieceCID     Size    Price            Duration  Verified  
...3q3uopfu  0       t01000    StorageDealFundsReserved  N          N         ...5qtonkdq  1016 B  0.000260639 FIL  521278    false     
...c2m26fvy  0       t01000    StorageDealClientFunding  N          N         ...a2mcy2gy  1016 B  0.000260558 FIL  521116    false  

5、check
lotus-miner sectors list
ID  State    OnChain  Active  Expiration                   Deals  DealWeight  VerifiedPower  
0   Proving  YES      YES     1550097 (in 10 weeks 1 day)  CC                                
1   Proving  YES      YES     1550097 (in 10 weeks 1 day)  1      2KiB        18KiB  

6、lotus-miner
lotus-miner storage-deals pending-publish
lotus-miner storage-deals pending-publish --publish-now

7、find
lotus client find <Data CID>

8、retrieve
lotus client retrieve <Data CID> <outfile>

log
lotus chain list --count 5
801: (Mar  8 04:25:51) [ bafy2bzacedxuocsnmgqdfnglvhbiq7rqqlljy65fvn4x2yfvcrhui2hjhqn5m: t01000, ]
802: (Mar  8 04:25:55) [ bafy2bzacec3nle6dk6syu7hv2zv5fc6vyekvrzejyjhyh7xdmsjbdpak63j2w: t01000, ]
803: (Mar  8 04:25:59) [ bafy2bzacebndptxinxpdbrn6pf525ufn5zvkqfscrb756dwj7pqa47dcfmwb4: t01000, ]
804: (Mar  8 04:26:03) [ bafy2bzaceai3pmwlxunfgqph6kkmftb4c757s7fvvy7mgur4mxtuxfumvrzvy: t01000, ]
805: (Mar  8 04:26:07) [ bafy2bzacea4xrtqn5sm3sywkje5xdnqi6vimithezmcubmwvybvkyyaj62rdo: t01000, ]
root@0d3872e3afdd:/home/lotus# head -c 450 /dev/urandom > 450_random_data.txt 
root@0d3872e3afdd:/home/lotus# lotus client import ./450_random_data.txt
Import 1646710431103465273, Root bafk2bzaced6dqiehjujukwcdr56o7z5oi6d5iuj6j54uueslcdut25ixf43jo
root@0d3872e3afdd:/home/lotus# lotus client local
1646710431103465272: bafk2bzaceaf3pam3lgllm5oddgwuzgu6auiqm2xzsirjbx6vwqfuq5mptp3jg @/home/lotus/512_random_data.txt (import)
1646710431103465273: bafk2bzaced6dqiehjujukwcdr56o7z5oi6d5iuj6j54uueslcdut25ixf43jo @/home/lotus/450_random_data.txt (import)
root@0d3872e3afdd:/home/lotus# lotus client deal bafk2bzaced6dqiehjujukwcdr56o7z5oi6d5iuj6j54uueslcdut25ixf43jo t01000 0.0000000005  518400
bafyreigovkhfbhy6yl3xeshm5f7d67stigyfwn6u67ybiddhtdc2m26fvy
root@0d3872e3afdd:/home/lotus# lotus client list-deals
DealCid      DealId  Provider  State                     On Chain?  Slashed?  PieceCID     Size    Price            Duration  Verified  
...3q3uopfu  0       t01000    StorageDealFundsReserved  N          N         ...5qtonkdq  1016 B  0.000260639 FIL  521278    false     
...c2m26fvy  0       t01000    StorageDealClientFunding  N          N         ...a2mcy2gy  1016 B  0.000260558 FIL  521116    false     
root@0d3872e3afdd:/home/lotus# lotus-miner sectors list
ID  State    OnChain  Active  Expiration                   Deals  DealWeight  VerifiedPower  
0   Proving  YES      YES     1550097 (in 10 weeks 1 day)  CC                                
1   Proving  YES      YES     1550097 (in 10 weeks 1 day)  1      2KiB        18KiB          
root@0d3872e3afdd:/home/lotus# lotus-miner storage-deals pending-publish
No deals queued to be published
root@0d3872e3afdd:/home/lotus# lotus-miner storage-deals pending-publish --publish-now
triggered deal publishing
root@0d3872e3afdd:/home/lotus# lotus client list-deals
DealCid      DealId  Provider  State                     On Chain?  Slashed?  PieceCID     Size    Price            Duration  Verified  
...3q3uopfu  0       t01000    StorageDealFundsReserved  N          N         ...5qtonkdq  1016 B  0.000260639 FIL  521278    false     
...c2m26fvy  0       t01000    StorageDealFundsReserved  N          N         ...a2mcy2gy  1016 B  0.000260558 FIL  521116    false     
root@0d3872e3afdd:/home/lotus# lotus client list-deals
DealCid      DealId  Provider  State                     On Chain?  Slashed?  PieceCID     Size    Price            Duration  Verified  
...3q3uopfu  0       t01000    StorageDealFundsReserved  N          N         ...5qtonkdq  1016 B  0.000260639 FIL  521278    false     
...c2m26fvy  0       t01000    StorageDealFundsReserved  N          N         ...a2mcy2gy  1016 B  0.000260558 FIL  521116    false     
root@0d3872e3afdd:/home/lotus# lotus chain list --count 5
836: (Mar  8 04:28:11) [ bafy2bzacebgp33fautf7ieeheyvzj3er4enlb2gp5ojezjqffglhjrzhymfa4: t01000, ]
837: (Mar  8 04:28:15) [ bafy2bzacect6oer3fiaazaihulayiwfcztf4icfqhsq7yi3nwb52shyjto2hi: t01000, ]
838: (Mar  8 04:28:19) [ bafy2bzacebvn6uj35yrn7iteshzcrvzj4xttrdq3r4lj5gq46eav6yktopzdq: t01000, ]
839: (Mar  8 04:28:23) [ bafy2bzacedk3lqr43s4qwj3pcsv3fufmwqz5vtlmlnudv262rg7tfoguiye4e: t01000, ]
840: (Mar  8 04:28:27) [ bafy2bzacedlacrnxc4sf3bz5w25nm4rbmfkxqi7shfhdxmxmppmmvysg25zb4: t01000, ]
root@0d3872e3afdd:/home/lotus# lotus client list-deals
DealCid      DealId  Provider  State                     On Chain?  Slashed?  PieceCID     Size    Price            Duration  Verified  
...3q3uopfu  0       t01000    StorageDealFundsReserved  N          N         ...5qtonkdq  1016 B  0.000260639 FIL  521278    false     
...c2m26fvy  0       t01000    StorageDealFundsReserved  N          N         ...a2mcy2gy  1016 B  0.000260558 FIL  521116    false     
root@0d3872e3afdd:/home/lotus# lotus client list-deals
DealCid      DealId  Provider  State                     On Chain?  Slashed?  PieceCID     Size    Price            Duration  Verified  
...3q3uopfu  0       t01000    StorageDealFundsReserved  N          N         ...5qtonkdq  1016 B  0.000260639 FIL  521278    false     
...c2m26fvy  0       t01000    StorageDealFundsReserved  N          N         ...a2mcy2gy  1016 B  0.000260558 FIL  521116    false     
root@0d3872e3afdd:/home/lotus# lotus chain list --count 5
841: (Mar  8 04:28:31) [ bafy2bzacebccyvzarktp7fb7rgjdl2asvym3ddt6e3nttythluhl4tl4ydjxo: t01000, ]
842: (Mar  8 04:28:35) [ bafy2bzacebba6zzr22vfutz4ogfip5cwsjayazcjxabx7ll7riquttwbm4cuc: t01000, ]
843: (Mar  8 04:28:39) [ bafy2bzacedxbij6bp3pmpy43rqixc2e6br5u2a7voptnavz3hblnuuovcf3n6: t01000, ]
844: (Mar  8 04:28:43) [ bafy2bzacecljctv3w6c6w6znhsr63evgiv7u4eavzyiaygob7io5omibgkvmg: t01000, ]
845: (Mar  8 04:28:47) [ bafy2bzaceakaopglcaloi5b4fnjpaw7qq6r3v54lw5vd5jue6uqlvg6mftd7m: t01000, ]

**Reply by alanmoleapfive** [reply]
Hi，Reiers
I have another question
try do it 
docker lotus Local mirror to build a multi-miner test environment devtest V1.14.4
first docker 
node +miner
two docker 
node +miner

first docker
1、start
docker run     -dit     --name first-lotus     -v /home/alan/firstlotus:/home/lotus     -p 1234:1234     -p 2345:2345     -p 3456:3456     -p 1777:1777     --entrypoint /bin/sh     192.168.1.45/blockchain/lotus:v1.14.4

2、into docker
  docker exec -it first-lotus /bin/bash
  cd /home/lotus
  rm -rf ~/.genesis-sectors

3、 Pre-seal some sectors for the genesis block
lotus-seed pre-seal --sector-size 2KiB --num-sectors 2

4、Create a genesis block
lotus-seed genesis new localnet.json


5、Create a default address and give it some funds
lotus-seed genesis add-miner localnet.json ~/.genesis-sectors/pre-seal-t01000.json

6、start node
   nohup lotus daemon --lotus-make-genesis=devgen.car --genesis-template=localnet.json --bootstrap=false &
     
7、set miner 
lotus wallet import --as-default ~/.genesis-sectors/pre-seal-t01000.key
out：
imported key t3xe5je75lkrvye32tfl37gug3az42iotuu3wxgkrhbpbvmum4lu26begiw74ju5a35nveqaw4ywdibj4y6kxq successfully!

8、Set up the genesis miner
lotus-miner init --genesis-miner --actor=t01000 --sector-size=2KiB --pre-sealed-sectors=~/.genesis-sectors --pre-sealed-metadata=~/.genesis-sectors/pre-seal-t01000.json --nosync

9、lotus-miner start
nohup lotus-miner run --nosync &

10、get listen
lotus net listen
/ip4/172.17.0.4/tcp/33189/p2p/12D3KooWH7s2kS4ZM4ofXX8egcJA1sb6UUju1QWxi3nVwhVGR2JH
/ip4/127.0.0.1/tcp/33189/p2p/12D3KooWH7s2kS4ZM4ofXX8egcJA1sb6UUju1QWxi3nVwhVGR2JH



two docker

1、start 
docker run     -dit     --name two-lotus     -v /home/alan/twolotus:/home/lotus     -p 21234:1234     -p 22345:2345     -p 23456:3456     -p 21777:1777     --entrypoint /bin/sh     192.168.1.45/blockchain/lotus:v1.14.2

2、copy devgen.car to 
cp /home/alan/firstlotus/devgen.car /home/alan/twolotus/

3、into docker
  docker exec -it two-lotus /bin/bash
  cd /home/lotus
  rm -rf ~/.genesis-sectors

4、start
nohup lotus daemon --genesis=devgen.car --bootstrap=false &

5、connect
lotus net connect /ip4/172.17.0.4/tcp/33189/p2p/12D3KooWH7s2kS4ZM4ofXX8egcJA1sb6UUju1QWxi3nVwhVGR2JH
connect 12D3KooWH7s2kS4ZM4ofXX8egcJA1sb6UUju1QWxi3nVwhVGR2JH: success


6、sync status
lotus sync status

sync status:
worker 0:
        Base:   [bafy2bzacedtphgr7bqouwflxvplijjyz2iqlruux7aj2wgvjg5xmycfjd5btw]
        Target: [bafy2bzacedwgukbssu3gi4xwzet4h7cvqgdyuyxidfd34rgghsbuw5jcwxcmy] (121)
        Height diff:    121
        Stage: message sync
        Height: 64
        Elapsed: 22.826980788s

this ok 
7、create BLS wallet
 lotus wallet new bls
t3uofdx5vvp7i43achugxdxmrggwgjwpykdeuwipvt4tc3m3t744vaulxqbg4wr4obz6b6j3lbxn2ic7fb7guq

8、Install 100 FIL from the genesis node to the current wallet
 lotus send t3uofdx5vvp7i43achugxdxmrggwgjwpykdeuwipvt4tc3m3t744vaulxqbg4wr4obz6b6j3lbxn2ic7fb7guq 100
bafy2bzaced75qy2rjwu3uyv32bzc47ewqdljm4fs2am4umcvzdw4grjpnzefu

9、check balance
 lotus wallet balance
100 FIL


 lotus wallet list
Address                                                                                 Balance  Nonce  Default
t3r33j64poh7x6o3obvhdtqtz5ychmyrdnh2caawfflmfpcdezx23vljpetcq7zp3gifpkdpe6hbzazw3ojp6a  100 FIL  0      X

10、Initialize a new node
lotus-miner init --sector-size=2KiB --nosync

this wait 
2022-02-22T07:58:19.155Z        INFO    main    lotus-miner/init.go:748 Pushed CreateMiner message: bafy2bzacebpcmr54hoz7ug6zo7e5okx6r44xhs7medbo2hdx4pzvyqd4t6bno
2022-02-22T07:58:19.155Z        INFO    main    lotus-miner/init.go:749 Waiting for confirmation
message not send to chain 
first mpool not has this message

11、 start lotus-miner
   nohup lotus-miner run &
   

12、info
   lotus-miner info

**Reply by LexLuthr** [reply]
@alanmoleapfive I would request you to open a separate discussion for running multiple miner. Discussions are driven by the topic they are focusing on. 

Please get me the output of `lotus client list-deals -v` from your lotus node. Also, please don't make the deal in a single command. Use the interactive CLI approach. It is easier to debug the problem this way.
Just run `lotus client deal` and then provide inputs one by one as required.

**Reply by alanmoleapfive** [reply]
lotus client list-deals -v
Created          DealCid                                                      DealId  Provider  State                     On Chain?  Slashed?  PieceCID                                                          Size    Price             Duration  TransferChannelID  TransferStatus  Verified  Message
Mar  9 08:36:22  bafyreigpoqck24mdsrs2dygtjgq6fgu5x6zorzk7gv7eh45n2zjhmpjuay  0       t01000    StorageDealFundsReserved  N          N         baga6ea4seaqnbbduraq5xcc62k5gm5yvoiyod6aml4dgvpfs5iqxa5ta2mcy2gy  1016 B  0.0002597415 FIL  519483

**Reply by LexLuthr** [reply]
Check if the message is on chain or not.

`lotus state search-msg bafyreigpoqck24mdsrs2dygtjgq6fgu5x6zorzk7gv7eh45n2zjhmpjuay` If the message is found on the chain then get me the following from the miner `lotus-miner storage-deals list -v` and `lotus-miner storage-deals pending-publish`

**Reply by alanmoleapfive** [reply]
lotus state search-msg bafyreigpoqck24mdsrs2dygtjgq6fgu5x6zorzk7gv7eh45n2zjhmpjuay 
ERROR: failed to load message: blockstore: block not found

root@0d3872e3afdd:/home/lotus# lotus client list-deals -v
Created          DealCid                                                      DealId  Provider  State                     On Chain?  Slashed?  PieceCID                                                          Size    Price             Duration  TransferChannelID  TransferStatus  Verified  Message
Mar  9 08:36:22  bafyreigpoqck24mdsrs2dygtjgq6fgu5x6zorzk7gv7eh45n2zjhmpjuay  0       t01000    StorageDealFundsReserved  N          N         baga6ea4seaqnbbduraq5xcc62k5gm5yvoiyod6aml4dgvpfs5iqxa5ta2mcy2gy  1016 B  0.0002597415 FIL  519483                                       false     
root@0d3872e3afdd:/home/lotus# lotus-miner storage-deals list -v
Creation  Verified  ProposalCid  DealId  State  Client  Size  Price  Duration  TransferChannelID  Message
root@0d3872e3afdd:/home/lotus# lotus-miner storage-deals pending-publish
No deals queued to be published
root@0d3872e3afdd:/home/lotus# lotus-miner storage-deals list -v
Creation  Verified  ProposalCid  DealId  State  Client  Size  Price  Duration  TransferChannelID  Message

**Reply by LexLuthr** [reply]
Okay. So your deal proposal in not part of the chain. That would explain the problem. Please check your local mpool and see if messages are stuck there or not. Either your miner is not configured properly to mine the blocks or mpool is not pushing the messages.

**Reply by alanmoleapfive** [reply]
this my env 
first docker
1、start
docker run -dit --name first-lotus -v /home/alan/firstlotus:/home/lotus -p 1234:1234 -p 2345:2345 -p 3456:3456 -p 1777:1777 --entrypoint /bin/sh 192.168.1.45/blockchain/lotus:v1.14.4

2、into docker
docker exec -it first-lotus /bin/bash
cd /home/lotus
rm -rf ~/.genesis-sectors

3、 Pre-seal some sectors for the genesis block
lotus-seed pre-seal --sector-size 2KiB --num-sectors 2

4、Create a genesis block
lotus-seed genesis new localnet.json

5、Create a default address and give it some funds
lotus-seed genesis add-miner localnet.json ~/.genesis-sectors/pre-seal-t01000.json

6、start node
nohup lotus daemon --lotus-make-genesis=devgen.car --genesis-template=localnet.json --bootstrap=false &

7、set miner
lotus wallet import --as-default ~/.genesis-sectors/pre-seal-t01000.key
out：
imported key t3xe5je75lkrvye32tfl37gug3az42iotuu3wxgkrhbpbvmum4lu26begiw74ju5a35nveqaw4ywdibj4y6kxq successfully!

8、Set up the genesis miner
lotus-miner init --genesis-miner --actor=t01000 --sector-size=2KiB --pre-sealed-sectors=/.genesis-sectors --pre-sealed-metadata=/.genesis-sectors/pre-seal-t01000.json --nosync

9、lotus-miner start
nohup lotus-miner run --nosync &


how do  configure

**Reply by LexLuthr** [reply]
@alanmoleapfive I would suggest running it outside of docker. Docker brings in a lot of network complexities. You steps look correct. Please try them outside of docker, directly on the base machine.

**Reply by alanmoleapfive** [reply]
not  use docker I try it  the same problem occurs

**Reply by alanmoleapfive** [reply]
You can also test the same version v1.14.4 make 2k

**Reply by LexLuthr** [reply]
I ran the test on 1.14.4 and I was able to make the deal without any problems. I would suggest you to check your system config and lotus config.

**Reply by alanmoleapfive** [reply]
system config and lotus config.  how to view
Can I see your operation process or operation log?

**Reply by LexLuthr** [reply]
Are you on filecoin slack. If not then please join https://filecoin.io/slack/ and reach out to me on #fil-lotus-help channel. We can do a screen share session and I can guide you with setting up 2k.

**Reply by alanmoleapfive** [reply]
ok

**Reply by alanmoleapfive** [reply]
thank you LexLuthr, problem solved, thank you for your remote assistance support
